js/build.js

```diff
@@ -0,0 +1,344 @@
+Fliplet.Widget.instance({
+  name: 'file-list',
+  displayName: 'File list',
+  icon: 'fa-exchange',
+  render: {
+    dependencies: [],
+    template: [
+      '<div data-view="content"></div>',
+    ].join(''),
+    ready: async function () {
+      var thisy = this;
+      const populateFileList = (entryID = Fliplet.Navigate.query.dataSourceEntryId, dsID = thisy.fields.dataSource) => {
+        var dataSourceId = dsID;
+        var dataSourceEntryId = entryID;
+        var columnName = thisy.fields.columnName; // TODO missing columName for list repeator
+        var type = thisy.fields.type;
+
+        if (!navigator.onLine) {
+          return Promise.reject('Please connect device to the internet')
+        } else if (!dataSourceEntryId) {
+          return Fliplet.UI.Toast('Missing dataSourceEntryId as a query parameter');
+        } else if (!dataSourceId) {
+          return Fliplet.UI.Toast('Please select Data Source from the Flie list component configuration');
+        } else if (!columnName) {
+          return Fliplet.UI.Toast('Please select Column Name from File list component configuration');
+        }
+
+        // TODO remove connect and findById
+        // pass record as parameter in the function when you have that informaion in the list repeator
+        // we have it for now only in record container
+        return Fliplet.DataSources.connect(dataSourceId).then(function (connection) {
+          return connection.findById(dataSourceEntryId).then(function (record) {
+            if (!checkArrayLinks(record.data[columnName])) {
+              $(document).find('[data-helper="file-list"]').html(`<p>There are no ${type}s</p>`)
+            } else {
+              var fileIDs = record.data[columnName].map(function (file) {
+                var url = typeof file === 'string'
```

Comment:
I would flip this because just because it's not a string doesn't mean it's an object. You don't want to encounter an error with `file.url` because it's not an object.

```suggestion
                var url = typeof file === 'object'
                  ? file.url
                  : file;
```

You then need to consider checking for what if `url` is not a string.